proc/proc.go

```diff
@@ -527,6 +545,9 @@ func (dbp *Process) SwitchGoroutine(gid int) error {
 // GoroutinesInfo returns an array of G structures representing the information
 // Delve cares about from the internal runtime G structure.
 func (dbp *Process) GoroutinesInfo() ([]*G, error) {
+	if dbp.exited {
+		return []*G{}, nil
```

Comment:
Since I was returning `[]*Threads{}, nil` for `ListThreads` I thought I would do the same thing here. I think returning `nil, &ProcessExitedError{}` is also fine if we do it for both. Your call.